### PR TITLE
Android results bar into advanced-search

### PR DESCRIPTION
**Pull request by JasBuc**

### CODE REVIEW

File: app/src/main/java/com/jetbrains/handson/mpp/mobile/MainActivity.kt [hunk 1]
```diff
@@ -32,12 +34,12 @@ class MainActivity : AppCompatActivity(), ApplicationContract.View {
...
 
-        val button = findViewById<Button>(R.id.button)
-        button.setOnClickListener {
+        val searchButton = findViewById<Button>(R.id.button)
+        searchButton.setOnClickListener {
             showLoadingSpinner(true)
             presenter.setNumAdults(numAdults)
             presenter.setNumChildren(numChildren)
-            if (departureTime=="Unset"){
+            if (departureTime=="Today, Now"){
```
Comment (beccar97) [inline app/src/main/java/com/jetbrains/handson/mpp/mobile/MainActivity.kt]: "Today, Now" seems to be turning into a bit of a magic string, do you think there's maybe a better way to handle the case that the user doesn't manually specify a time?

File: app/src/main/java/com/jetbrains/handson/mpp/mobile/AdvancedSearchActivity.kt [hunk 2]
```diff
@@ -93,15 +94,16 @@ class AdvancedSearchActivity : AppCompatActivity(), AdvancedSearchContract.View,
...
-            intent.putExtra("time", getDateInAPIFormat())
-        } else {
-            intent.putExtra("time", "Unset")
-        }
+        intent.putExtra("time", getDateInAPIFormat())
+        intent.putExtra("timeText", getTimeText())
         setResult(2, intent)
         finish()
     }
     private fun getDateInAPIFormat(): String {
```
Comment (beccar97) [inline app/src/main/java/com/jetbrains/handson/mpp/mobile/AdvancedSearchActivity.kt:94]: There's some commonality between this method and the `getTimeText` method, maybe you could have a method called something like `getTime` that returns a DateTime object, which both of these functions can use, and then can use different DateTimeFormats to create the strings you need?

File: app/src/main/java/com/jetbrains/handson/mpp/mobile/MainActivity.kt (hunk 1 above)
Comment (JasBuc) [reply app/src/main/java/com/jetbrains/handson/mpp/mobile/MainActivity.kt]: This has been changed now and implemented using a new getTime function as you mentioned in the comment below

File: app/src/main/java/com/jetbrains/handson/mpp/mobile/AdvancedSearchActivity.kt [hunk 3]
```diff
@@ -103,7 +103,7 @@ class AdvancedSearchActivity : AppCompatActivity(), AdvancedSearchContract.View,
         if (!timeChanged){
             return DateTimeTz.nowLocal().format(dateTimeFormat)
         }
-        val newDateTime = DateTime(myYear, myMonth, myDay, myHour, myMinute)
+        val newDateTime = DateTime(year, month, day, hour, minute)
         return newDateTime.format(dateTimeFormat)
```
Comment (beccar97) [inline app/src/main/java/com/jetbrains/handson/mpp/mobile/AdvancedSearchActivity.kt:107]: Not a big deal, but you're doing the formatting twice, you could simplify by doing
```kotlin
val dateTime = if (timeChanged) DateTime(year, month, day, hour, minute) else DateTimeTz.nowLocal()
return DateTime.format(dateTimeFormat)
```

File: app/src/main/java/com/jetbrains/handson/mpp/mobile/MainActivity.kt [hunk 4]
```diff
@@ -46,15 +46,19 @@ class MainActivity : AppCompatActivity(), ApplicationContract.View {
 
         val adultsButton = findViewById<Button>(R.id.selected_adults)
         adultsButton.setOnClickListener() {
-            presenter.setNumAdults(0)
-            adultsButton.visibility = View.INVISIBLE
-            searchButton.performClick()
+            if (presenter.canClearAdults()) {
```
Comment (beccar97) [inline app/src/main/java/com/jetbrains/handson/mpp/mobile/MainActivity.kt:49]: Nice, this is a good check to have!

File: SharedCode/src/commonMain/kotlin/ApplicationPresenter.kt [hunk 5]
```diff
@@ -72,6 +70,11 @@ class ApplicationPresenter: ApplicationContract.Presenter() {
         }
     }
 
+    override fun getTimeNow(): String {
```
Comment (beccar97) [inline SharedCode/src/commonMain/kotlin/ApplicationPresenter.kt:73]: Is this function ever actually called from the view? Does it need to be on the contract?

Comment (JasBuc) [reply SharedCode/src/commonMain/kotlin/ApplicationPresenter.kt:73]: It's used when the advanced search time is cleared to reset the time to now

**Review by beccar97 — APPROVED** [review]
Looks good 😄

**Pull request merged**